### PR TITLE
[HEOS] Update pause/stop handling

### PR DESCRIPTION
**Pull request by martinvw**
Update PAUSE/STOP handling depending on the item that is currently playing.

As discussed in https://community.openhab.org/t/heos-denon-support/3449/379?u=martinvw

I must admit I did not test in my OH3 install, will try that later today

The default handling of pause vs stop does not work consistently between online-radio and music from your network, this PR aims to improve that handling.

You can download a test-version at: https://download.martinvw.nl/org.openhab.binding.heos-3.1.0-SNAPSHOT-stop-pause.jar

Signed-off-by: Martin van Wingerden <martin@martinvw.nl>

### CODE REVIEW

**Comment by Wire82** [comment]
@martinvw I checked the code and from my perspective I saw no issue. But I also haven't tested it. An interesting thing is, that I never saw that kind of issue. But maybe this was just luck. 😄

**Comment by martinvw** [comment]
Okay tested :-)

File: bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/handler/HeosChannelHandlerControl.java [hunk 1]
```diff
@@ -79,4 +92,28 @@ private void handleCommand(Command command, String id) throws IOException, ReadE
...
     }
+
+    private boolean shouldPause(String id) {
+        Media applicableMedia = playingMediaCache.get(id);
+        if (applicableMedia == null || SONG.equals(applicableMedia.type)) {
+            return true;
+        } else {
+            // we have a station here, just have to check which one
+            switch (applicableMedia.sourceId) {
+                case 3: // Tunein
```
Comment (Wire82) [inline bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/handler/HeosChannelHandlerControl.java:103]: Does it make sense in later changes to define the different Media Types (source IDs) as an Constant or ENUM? Just a question for the future.

**Review by Wire82 — APPROVED** [review]

**Comment by martinvw** [comment]
Force pushed because of compile error on master

File: bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/handler/HeosChannelHandlerControl.java (hunk 1 above)
Comment (martinvw) [reply bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/handler/HeosChannelHandlerControl.java]: Yes, that makes a lot of sense, good idea, I updated it to use constants.

**Review by kaikreuzer — APPROVED** [review]
Thanks!

**Pull request merged**